### ISSUE TITLE
Deletes babel code, leading to broken while loops

### ISSUE THREAD

**Issue by mitchellwarr** [rollup/rollup#4147]
- Rollup Version: 2.52.2
- Operating System (or Browser): Mac
- Node Version (if applicable): v12.19.0
- Link to reproduction: [reproduction](https://rollupjs.org/repl/?version=2.52.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMF9yZWdlbmVyYXRvclJ1bnRpbWUlMjBmcm9tJTIwJTVDJTIyJTQwYmFiZWwlMkZydW50aW1lJTJGcmVnZW5lcmF0b3IlNUMlMjIlM0IlNUNuaW1wb3J0JTIwJTVDJTIyY29yZS1qcyUyRm1vZHVsZXMlMkZlcy5vYmplY3QudG8tc3RyaW5nLmpzJTVDJTIyJTNCJTVDbmltcG9ydCUyMCU1QyUyMmNvcmUtanMlMkZtb2R1bGVzJTJGZXMucHJvbWlzZS5qcyU1QyUyMiUzQiU1Q25pbXBvcnQlMjAlNUMlMjJjb3JlLWpzJTJGbW9kdWxlcyUyRndlYi50aW1lcnMuanMlNUMlMjIlM0IlNUNuJTVDbmZ1bmN0aW9uJTIwYXN5bmNHZW5lcmF0b3JTdGVwKGdlbiUyQyUyMHJlc29sdmUlMkMlMjByZWplY3QlMkMlMjBfbmV4dCUyQyUyMF90aHJvdyUyQyUyMGtleSUyQyUyMGFyZyklMjAlN0IlMjB0cnklMjAlN0IlMjB2YXIlMjBpbmZvJTIwJTNEJTIwZ2VuJTVCa2V5JTVEKGFyZyklM0IlMjB2YXIlMjB2YWx1ZSUyMCUzRCUyMGluZm8udmFsdWUlM0IlMjAlN0QlMjBjYXRjaCUyMChlcnJvciklMjAlN0IlMjByZWplY3QoZXJyb3IpJTNCJTIwcmV0dXJuJTNCJTIwJTdEJTIwaWYlMjAoaW5mby5kb25lKSUyMCU3QiUyMHJlc29sdmUodmFsdWUpJTNCJTIwJTdEJTIwZWxzZSUyMCU3QiUyMFByb21pc2UucmVzb2x2ZSh2YWx1ZSkudGhlbihfbmV4dCUyQyUyMF90aHJvdyklM0IlMjAlN0QlMjAlN0QlNUNuJTVDbmZ1bmN0aW9uJTIwX2FzeW5jVG9HZW5lcmF0b3IoZm4pJTIwJTdCJTIwcmV0dXJuJTIwZnVuY3Rpb24lMjAoKSUyMCU3QiUyMHZhciUyMHNlbGYlMjAlM0QlMjB0aGlzJTJDJTIwYXJncyUyMCUzRCUyMGFyZ3VtZW50cyUzQiUyMHJldHVybiUyMG5ldyUyMFByb21pc2UoZnVuY3Rpb24lMjAocmVzb2x2ZSUyQyUyMHJlamVjdCklMjAlN0IlMjB2YXIlMjBnZW4lMjAlM0QlMjBmbi5hcHBseShzZWxmJTJDJTIwYXJncyklM0IlMjBmdW5jdGlvbiUyMF9uZXh0KHZhbHVlKSUyMCU3QiUyMGFzeW5jR2VuZXJhdG9yU3RlcChnZW4lMkMlMjByZXNvbHZlJTJDJTIwcmVqZWN0JTJDJTIwX25leHQlMkMlMjBfdGhyb3clMkMlMjAlNUMlMjJuZXh0JTVDJTIyJTJDJTIwdmFsdWUpJTNCJTIwJTdEJTIwZnVuY3Rpb24lMjBfdGhyb3coZXJyKSUyMCU3QiUyMGFzeW5jR2VuZXJhdG9yU3RlcChnZW4lMkMlMjByZXNvbHZlJTJDJTIwcmVqZWN0JTJDJTIwX25leHQlMkMlMjBfdGhyb3clMkMlMjAlNUMlMjJ0aHJvdyU1QyUyMiUyQyUyMGVyciklM0IlMjAlN0QlMjBfbmV4dCh1bmRlZmluZWQpJTNCJTIwJTdEKSUzQiUyMCU3RCUzQiUyMCU3RCU1Q24lNUNuaW1wb3J0JTIwJTdCJTIwdXNlRWZmZWN0JTJDJTIwdXNlUmVmJTIwJTdEJTIwZnJvbSUyMCdyZWFjdCclM0IlNUNuJTVDbnZhciUyMGludGVydmFsJTIwJTNEJTIwZnVuY3Rpb24lMjBpbnRlcnZhbChmJTJDJTIwZGVsYXkpJTIwJTdCJTVDbiUyMCUyMHZhciUyMHglMjAlM0QlMjAlN0IlNUNuJTIwJTIwJTIwJTIwbG9vcGluZyUzQSUyMHRydWUlMkMlNUNuJTIwJTIwJTIwJTIwYWJvcnQlM0ElMjBmdW5jdGlvbiUyMGFib3J0KCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwdGhpcy5sb29waW5nJTIwJTNEJTIwZmFsc2UlM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCU3RCUzQiU1Q24lNUNuJTIwJTIwdmFyJTIwbG9vcCUyMCUzRCUyMCUyRiolMjNfX1BVUkVfXyolMkZmdW5jdGlvbiUyMCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHZhciUyMF9yZWYlMjAlM0QlMjBfYXN5bmNUb0dlbmVyYXRvciglMjAlMkYqJTIzX19QVVJFX18qJTJGX3JlZ2VuZXJhdG9yUnVudGltZS5tYXJrKGZ1bmN0aW9uJTIwX2NhbGxlZSgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMHJldHVybiUyMF9yZWdlbmVyYXRvclJ1bnRpbWUud3JhcChmdW5jdGlvbiUyMF9jYWxsZWUlMjQoX2NvbnRleHQpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHdoaWxlJTIwKDEpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHN3aXRjaCUyMChfY29udGV4dC5wcmV2JTIwJTNEJTIwX2NvbnRleHQubmV4dCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwY2FzZSUyMDAlM0ElNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwaWYlMjAoIXgubG9vcGluZyklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwX2NvbnRleHQubmV4dCUyMCUzRCUyMDclM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwYnJlYWslM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTdEJTVDbiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBfY29udGV4dC5uZXh0JTIwJTNEJTIwMyUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBmKCklM0IlNUNuJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGNhc2UlMjAzJTNBJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMF9jb250ZXh0Lm5leHQlMjAlM0QlMjA1JTNCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHJldHVybiUyMG5ldyUyMFByb21pc2UoZnVuY3Rpb24lMjAocmVzb2x2ZSklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwc2V0VGltZW91dChyZXNvbHZlJTJDJTIwZGVsYXkpJTNCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU3RCklM0IlNUNuJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGNhc2UlMjA1JTNBJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMF9jb250ZXh0Lm5leHQlMjAlM0QlMjAwJTNCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGJyZWFrJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjYXNlJTIwNyUzQSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjYXNlJTIwJTVDJTIyZW5kJTVDJTIyJTNBJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHJldHVybiUyMF9jb250ZXh0LnN0b3AoKSUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlN0QlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCU3RCUyQyUyMF9jYWxsZWUpJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCkpJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjByZXR1cm4lMjBmdW5jdGlvbiUyMGxvb3AoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBfcmVmLmFwcGx5KHRoaXMlMkMlMjBhcmd1bWVudHMpJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCUzQiU1Q24lMjAlMjAlN0QoKSUzQiU1Q24lNUNuJTIwJTIwbG9vcCgpJTNCJTVDbiUyMCUyMHJldHVybiUyMHglM0IlNUNuJTdEJTNCJTVDbiU1Q252YXIlMjB1c2VJbnRlcnZhbCUyMCUzRCUyMGZ1bmN0aW9uJTIwdXNlSW50ZXJ2YWwoY2FsbGJhY2slMkMlMjBkZWxheSklMjAlN0IlNUNuJTIwJTIwdmFyJTIwc2F2ZWRDYWxsYmFjayUyMCUzRCUyMHVzZVJlZigpJTNCJTVDbiUyMCUyMHNhdmVkQ2FsbGJhY2suY3VycmVudCUyMCUzRCUyMGNhbGxiYWNrJTNCJTVDbiUyMCUyMHVzZUVmZmVjdChmdW5jdGlvbiUyMCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHZhciUyMHRpY2slMjAlM0QlMjBmdW5jdGlvbiUyMHRpY2soKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBzYXZlZENhbGxiYWNrLmN1cnJlbnQoKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlM0IlNUNuJTVDbiUyMCUyMCUyMCUyMGlmJTIwKGRlbGF5JTIwISUzRCUzRCUyMG51bGwlMjAlMjYlMjYlMjBkZWxheSUyMCElM0QlM0QlMjB1bmRlZmluZWQpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMHZhciUyMGNsZWFyJTIwJTNEJTIwaW50ZXJ2YWwodGljayUyQyUyMGRlbGF5KSUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBmdW5jdGlvbiUyMCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMGNsZWFyLmFib3J0KCklM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTdEJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lMjAlMjAlN0QlMkMlMjAlNUJkZWxheSU1RCklM0IlNUNuJTdEJTNCJTVDbiU1Q25leHBvcnQlMjBkZWZhdWx0JTIwdXNlSW50ZXJ2YWwlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

### Expected Behavior

The main code should remain the same between both pieces

### Actual Behavior

Lines 26:29 of the input babel transpiled code gets removed
```js
if (!x.looping) {
  _context.next = 7;
  break;
}
```
This might be due to dead code removal, but this isnt a dead end branch! In fact its quite important to stop an infinite loop!

![image](https://user-images.githubusercontent.com/20696187/123345646-a5ef1d80-d5aa-11eb-80a4-8f7ca6881026.png)

### The untranspiled code
```js
import { useEffect, useRef } from 'react';

const interval = (f, delay) => {
  let x = {
    looping: true,
    abort() {
      this.looping = false;
    }
  };
  let loop = async function() {
    while (x.looping) {
      await f();
      await new Promise(resolve => setTimeout(resolve, delay));
    }
  };
  loop();
  return x;
};

const useInterval = (callback, delay) => {
  const savedCallback = useRef();
  savedCallback.current = callback;

  useEffect(
    () => {
      const tick = () => savedCallback.current();

      if (delay !== null && delay !== undefined) {
        let clear = interval(tick, delay);
        return () => {
          clear.abort();
        };
      }
    },
    [delay]
  );
};

export default useInterval;
```

### Workaround
[Here is a slight change that makes it work](https://rollupjs.org/repl/?version=2.52.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMF9yZWdlbmVyYXRvclJ1bnRpbWUlMjBmcm9tJTIwJTVDJTIyJTQwYmFiZWwlMkZydW50aW1lJTJGcmVnZW5lcmF0b3IlNUMlMjIlM0IlNUNuaW1wb3J0JTIwJTVDJTIyY29yZS1qcyUyRm1vZHVsZXMlMkZlcy5vYmplY3QudG8tc3RyaW5nLmpzJTVDJTIyJTNCJTVDbmltcG9ydCUyMCU1QyUyMmNvcmUtanMlMkZtb2R1bGVzJTJGZXMucHJvbWlzZS5qcyU1QyUyMiUzQiU1Q25pbXBvcnQlMjAlNUMlMjJjb3JlLWpzJTJGbW9kdWxlcyUyRndlYi50aW1lcnMuanMlNUMlMjIlM0IlNUNuJTVDbmZ1bmN0aW9uJTIwYXN5bmNHZW5lcmF0b3JTdGVwKGdlbiUyQyUyMHJlc29sdmUlMkMlMjByZWplY3QlMkMlMjBfbmV4dCUyQyUyMF90aHJvdyUyQyUyMGtleSUyQyUyMGFyZyklMjAlN0IlMjB0cnklMjAlN0IlMjB2YXIlMjBpbmZvJTIwJTNEJTIwZ2VuJTVCa2V5JTVEKGFyZyklM0IlMjB2YXIlMjB2YWx1ZSUyMCUzRCUyMGluZm8udmFsdWUlM0IlMjAlN0QlMjBjYXRjaCUyMChlcnJvciklMjAlN0IlMjByZWplY3QoZXJyb3IpJTNCJTIwcmV0dXJuJTNCJTIwJTdEJTIwaWYlMjAoaW5mby5kb25lKSUyMCU3QiUyMHJlc29sdmUodmFsdWUpJTNCJTIwJTdEJTIwZWxzZSUyMCU3QiUyMFByb21pc2UucmVzb2x2ZSh2YWx1ZSkudGhlbihfbmV4dCUyQyUyMF90aHJvdyklM0IlMjAlN0QlMjAlN0QlNUNuJTVDbmZ1bmN0aW9uJTIwX2FzeW5jVG9HZW5lcmF0b3IoZm4pJTIwJTdCJTIwcmV0dXJuJTIwZnVuY3Rpb24lMjAoKSUyMCU3QiUyMHZhciUyMHNlbGYlMjAlM0QlMjB0aGlzJTJDJTIwYXJncyUyMCUzRCUyMGFyZ3VtZW50cyUzQiUyMHJldHVybiUyMG5ldyUyMFByb21pc2UoZnVuY3Rpb24lMjAocmVzb2x2ZSUyQyUyMHJlamVjdCklMjAlN0IlMjB2YXIlMjBnZW4lMjAlM0QlMjBmbi5hcHBseShzZWxmJTJDJTIwYXJncyklM0IlMjBmdW5jdGlvbiUyMF9uZXh0KHZhbHVlKSUyMCU3QiUyMGFzeW5jR2VuZXJhdG9yU3RlcChnZW4lMkMlMjByZXNvbHZlJTJDJTIwcmVqZWN0JTJDJTIwX25leHQlMkMlMjBfdGhyb3clMkMlMjAlNUMlMjJuZXh0JTVDJTIyJTJDJTIwdmFsdWUpJTNCJTIwJTdEJTIwZnVuY3Rpb24lMjBfdGhyb3coZXJyKSUyMCU3QiUyMGFzeW5jR2VuZXJhdG9yU3RlcChnZW4lMkMlMjByZXNvbHZlJTJDJTIwcmVqZWN0JTJDJTIwX25leHQlMkMlMjBfdGhyb3clMkMlMjAlNUMlMjJ0aHJvdyU1QyUyMiUyQyUyMGVyciklM0IlMjAlN0QlMjBfbmV4dCh1bmRlZmluZWQpJTNCJTIwJTdEKSUzQiUyMCU3RCUzQiUyMCU3RCU1Q24lNUNuaW1wb3J0JTIwJTdCJTIwdXNlRWZmZWN0JTJDJTIwdXNlUmVmJTIwJTdEJTIwZnJvbSUyMCdyZWFjdCclM0IlNUNuJTVDbnZhciUyMGludGVydmFsJTIwJTNEJTIwZnVuY3Rpb24lMjBpbnRlcnZhbChmJTJDJTIwZGVsYXkpJTIwJTdCJTVDbiUyMCUyMHZhciUyMHglMjAlM0QlMjAlN0IlNUNuJTIwJTIwJTIwJTIwbG9vcGluZyUzQSUyMHRydWUlMkMlNUNuJTIwJTIwJTIwJTIwYWJvcnQlM0ElMjBmdW5jdGlvbiUyMGFib3J0KCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwdGhpcy5sb29waW5nJTIwJTNEJTIwZmFsc2UlM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCU3RCUzQiU1Q24lNUN0eC5sb29waW5nJTIwJTNEJTIwdHJ1ZSUzQiU1Q24lNUNuJTIwJTIwdmFyJTIwbG9vcCUyMCUzRCUyMCUyRiolMjNfX1BVUkVfXyolMkZmdW5jdGlvbiUyMCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMHZhciUyMF9yZWYlMjAlM0QlMjBfYXN5bmNUb0dlbmVyYXRvciglMjAlMkYqJTIzX19QVVJFX18qJTJGX3JlZ2VuZXJhdG9yUnVudGltZS5tYXJrKGZ1bmN0aW9uJTIwX2NhbGxlZSgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMHJldHVybiUyMF9yZWdlbmVyYXRvclJ1bnRpbWUud3JhcChmdW5jdGlvbiUyMF9jYWxsZWUlMjQoX2NvbnRleHQpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHdoaWxlJTIwKDEpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHN3aXRjaCUyMChfY29udGV4dC5wcmV2JTIwJTNEJTIwX2NvbnRleHQubmV4dCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwY2FzZSUyMDAlM0ElNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwaWYlMjAoIXgubG9vcGluZyklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwX2NvbnRleHQubmV4dCUyMCUzRCUyMDclM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTNCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU3RCU1Q24lNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwX2NvbnRleHQubmV4dCUyMCUzRCUyMDMlM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwZigpJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjYXNlJTIwMyUzQSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBfY29udGV4dC5uZXh0JTIwJTNEJTIwNSUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBuZXclMjBQcm9taXNlKGZ1bmN0aW9uJTIwKHJlc29sdmUpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMHJldHVybiUyMHNldFRpbWVvdXQocmVzb2x2ZSUyQyUyMGRlbGF5KSUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlN0QpJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjYXNlJTIwNSUzQSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBfY29udGV4dC5uZXh0JTIwJTNEJTIwMCUzQiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBicmVhayUzQiU1Q24lNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwY2FzZSUyMDclM0ElNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwY2FzZSUyMCU1QyUyMmVuZCU1QyUyMiUzQSU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjByZXR1cm4lMjBfY29udGV4dC5zdG9wKCklM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCUyMCUyMCU3RCU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlN0QlMkMlMjBfY2FsbGVlKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QpKSUzQiU1Q24lNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwZnVuY3Rpb24lMjBsb29wKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwX3JlZi5hcHBseSh0aGlzJTJDJTIwYXJndW1lbnRzKSUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlM0IlNUNuJTIwJTIwJTdEKCklM0IlNUNuJTVDbiUyMCUyMGxvb3AoKSUzQiU1Q24lMjAlMjByZXR1cm4lMjB4JTNCJTVDbiU3RCUzQiU1Q24lNUNudmFyJTIwdXNlSW50ZXJ2YWwlMjAlM0QlMjBmdW5jdGlvbiUyMHVzZUludGVydmFsKGNhbGxiYWNrJTJDJTIwZGVsYXkpJTIwJTdCJTVDbiUyMCUyMHZhciUyMHNhdmVkQ2FsbGJhY2slMjAlM0QlMjB1c2VSZWYoKSUzQiU1Q24lMjAlMjBzYXZlZENhbGxiYWNrLmN1cnJlbnQlMjAlM0QlMjBjYWxsYmFjayUzQiU1Q24lMjAlMjB1c2VFZmZlY3QoZnVuY3Rpb24lMjAoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjB2YXIlMjB0aWNrJTIwJTNEJTIwZnVuY3Rpb24lMjB0aWNrKCklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwc2F2ZWRDYWxsYmFjay5jdXJyZW50KCklM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTNCJTVDbiU1Q24lMjAlMjAlMjAlMjBpZiUyMChkZWxheSUyMCElM0QlM0QlMjBudWxsJTIwJTI2JTI2JTIwZGVsYXklMjAhJTNEJTNEJTIwdW5kZWZpbmVkKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjB2YXIlMjBjbGVhciUyMCUzRCUyMGludGVydmFsKHRpY2slMkMlMjBkZWxheSklM0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwZnVuY3Rpb24lMjAoKSUyMCU3QiU1Q24lMjAlMjAlMjAlMjAlMjAlMjAlMjAlMjBjbGVhci5hYm9ydCgpJTNCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMCU3RCUzQiU1Q24lMjAlMjAlMjAlMjAlN0QlNUNuJTIwJTIwJTdEJTJDJTIwJTVCZGVsYXklNUQpJTNCJTVDbiU3RCUzQiU1Q24lNUNuZXhwb3J0JTIwZGVmYXVsdCUyMHVzZUludGVydmFsJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==)

Essentially it adds this one line
```js
var x = {
  looping: true,
  abort: function abort() {
    this.looping = false;
  }
};
x.looping = true; // This line
```

After that, it all works

**Comment by lukastaegert** [comment]
Thanks for posting, this looks like a serious bug. One thing I can tell you so far: The problem where Rollup is struggling is actually not the  Babel code, that one just avoids some necessary deoptimizations Rollup has in place currently for async functions. It is this line:

```js
this.looping = false;
```

If you replace it with

```js
x.looping = false;
```

then it works, i.e. Rollup fails to acknowledge that `this`  is likely `x` or otherwise `x` needs to be deoptimized in some way. I will have a look, but at least this should give you a better workaround for now.

**Comment by lukastaegert** [comment]
It seems to be something really subtle. For my own and other's reference, this is as much as I could boil down the reproduction for now:

https://rollupjs.org/repl/?version=2.52.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMGRvU3R1ZmYlMjBmcm9tJTIwJ2V4dGVybmFsJyUzQiU1Q24lNUNudmFyJTIwaW50ZXJ2YWwlMjAlM0QlMjAoKSUyMCUzRCUzRSUyMCU3QiU1Q24lMjAlMjB2YXIlMjB4JTIwJTNEJTIwJTdCJTVDbiUyMCUyMCUyMCUyMGxvb3BpbmclM0ElMjB0cnVlJTJDJTVDbiUyMCUyMCUyMCUyMGFib3J0JTNBJTIwZnVuY3Rpb24lMjBhYm9ydCgpJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyMCUyMHRoaXMubG9vcGluZyUyMCUzRCUyMGZhbHNlJTNCJTVDbiUyMCUyMCUyMCUyMCU3RCU1Q24lMjAlMjAlN0QlM0IlNUNuJTVDdCU1Q24lNUN0ZG9TdHVmZigoKSUyMCUzRCUzRSUyMCU3QiU1Q24lNUN0JTVDdCUyRiUyRiUyMFRoaXMlMjBuZWVkcyUyMHRvJTIwYmUlMjByZXRhaW5lZCU1Q24lNUN0JTIwJTIwaWYlMjAoIXgubG9vcGluZyklMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIwJTIwcmV0dXJuJTIwZmFsc2UlM0IlNUNuJTIwJTIwJTIwJTIwJTdEJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMHRydWUlM0IlNUNuJTIwJTIwJTdEKSU1Q24lNUNuJTIwJTIwcmV0dXJuJTIweCUzQiU1Q24lN0QlM0IlNUNuJTVDbmV4cG9ydCUyMGRlZmF1bHQlMjAoKSUyMCUzRCUzRSUyMCU3QiU1Q24lNUN0JTJGJTJGJTIwSW5saW5pbmclMjB0aGUlMjBpZiUyMHNlZW1zJTIwdG8lMjBmaXglMjBpdCU1Q24lMjAlMjBpZiUyMCh0cnVlKSUyMCU3QiU1Q24lNUN0JTVDdCUyRiUyRiUyMFVzaW5nJTIwY29uc3QlMjBzZWVtcyUyMHRvJTIwZml4JTIwaXQlNUNuJTIwJTIwJTIwJTIwdmFyJTIwY2xlYXIlMjAlM0QlMjBpbnRlcnZhbCgpJTNCJTVDbiUyMCUyMCUyMCUyMHJldHVybiUyMCgpJTIwJTNEJTNFJTIwY2xlYXIuYWJvcnQoKSUzQiU1Q24lMjAlMjAlN0QlNUNuJTdEJTNCJTVDbiUyMiUyQyUyMmlzRW50cnklMjIlM0F0cnVlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=

**Comment by lukastaegert** [comment]
Turns out the actual issue was very different from what I originally thought. Fix at #4149

**Comment by mitchellwarr** [comment]
Thank you for the quick fix!
